src/components/document/document.tsx

```diff
@@ -454,20 +466,14 @@ export class DocumentComponent extends BaseComponent<IProps, IState> {
       });
   }
 
-  private handlePublishWorkspace = () => {
-    const { document } = this.props;
-    const { db, ui, appConfig } = this.stores;
-    const docTypeString = appConfig.getDocumentLabel(document.type, 1);
-    // TODO: Disable publish button while publishing
-    db.publishProblemDocument(document)
-      .then(() => ui.alert(`Your ${docTypeString} was published.`, `${docTypeString} Published`));
+  private handlePublishSupport = () => {
+    const { document, onPublishSupport } = this.props;
+    onPublishSupport && onPublishSupport(document);
```

Comment:
The almost released TypeScript 3.7 is going to allow for conditional chaining so this will be able to written as `onPublishSupport?.(document)` once it is released.  I'm not sure why the decided they needed the dot as CoffeeScript didn't need it - it must be a parser issue.

Reply:
From https://github.com/tc39/proposal-optional-chaining#faq:

**`obj?.[expr]` and `func?.(arg)` look ugly. Why not use `obj?[expr]` and `func?(arg)` as does {language X}?**

We don’t use the `obj?[expr]` and `func?(arg)` syntax, because of the difficulty for the parser to efficiently distinguish those forms from the conditional operator, e.g., `obj?[expr].filter(fun):0` and `func?(x - 2) + 3 :1`.